save-demo/src/main/kotlin/com/saveourtool/save/demo/service/KubernetesService.kt

```diff
@@ -0,0 +1,137 @@
+package com.saveourtool.save.demo.service
+
+import com.saveourtool.save.demo.DemoAgentConfig
+import com.saveourtool.save.demo.DemoStatus
+import com.saveourtool.save.demo.config.ConfigProperties
+import com.saveourtool.save.demo.entity.Demo
+import com.saveourtool.save.demo.utils.*
+import com.saveourtool.save.utils.*
+
+import io.fabric8.kubernetes.api.model.*
+import io.fabric8.kubernetes.client.KubernetesClient
+import io.ktor.client.*
+import io.ktor.client.engine.apache.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.server.ResponseStatusException
+import reactor.core.publisher.Mono
+
+import java.net.ConnectException
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.serialization.json.Json
+
+/**
+ * Service for interactions with kubernetes
+ */
+@Service
+class KubernetesService(
+    private val kc: KubernetesClient,
+    private val configProperties: ConfigProperties,
+) {
+    private val kubernetesSettings = configProperties.kubernetes
+
+    /**
+     * @param demo demo entity
+     * @param version version of [demo] that should be run in kubernetes pod
+     * @return [Mono] of [StringResponse] filled with readable message
+     */
+    @Suppress("TOO_MANY_LINES_IN_LAMBDA")
+    fun start(demo: Demo, version: String = "manual"): Mono<StringResponse> = Mono.create { sink ->
```

Comment:
why you need a sink?
does work a default implementation:
```kotlin
fun start(demo: Demo, version: String = "manual"): Mono<StringResponse> = Mono.fromCallable {
        logger.info("Creating job ${jobNameForDemo(demo)}...")
        try {
            kc.startJob(demo, kubernetesSettings)
            demo
        } catch (kre: KubernetesRunnerException) {
            throw ResponseStatusException(
                HttpStatus.INTERNAL_SERVER_ERROR,
                "Could not create job for ${jobNameForDemo(demo)}",
                kre,
            )
        }
    }
```

Reply:
Done